### PR TITLE
Plugins: Update the Install Plugin component to match the activate toggle component

### PR DESCRIPTION
**Pull request by enejb**
Similar to #10223
I am just updating the code to be more in line with the activate toggle component. 

### To Test
Install a plugin from single plugin view and the single plugin single site view.

### CODE REVIEW

File: client/my-sites/plugins/plugin-install-button/index.jsx [hunk 1]
```diff
@@ -184,22 +220,42 @@ module.exports = React.createClass( {
...
+		if ( ! selectedSite.canUpdateFiles ) {
 			return this.renderDisabledNotice();
 		}
 
 		return this.renderButton();
 	}
-} );
+}
+
+PluginInstallButton.displayName = 'PluginInstallButton';
```
Comment (tyxla) [inline client/my-sites/plugins/plugin-install-button/index.jsx]: This line is no longer necessary, as `displayName` will default to the class name, which is essentially the same.

File: client/my-sites/plugins/plugin-install-button/index.jsx [hunk 2]
```diff
@@ -184,22 +220,42 @@ module.exports = React.createClass( {
...
+PluginInstallButton.propTypes = {
```
Comment (tyxla) [inline client/my-sites/plugins/plugin-install-button/index.jsx]: Does it make sense to add default values for non-required props? This is not necessarily related with this PR, so feel free to ignore.

File: client/my-sites/plugins/plugin-install-button/index.jsx [hunk 3]
```diff
@@ -184,22 +220,42 @@ module.exports = React.createClass( {
...
+	disabled: React.PropTypes.bool,
+};
+
+export default connect(
+	null,
+	{
+		recordGoogleEvent,
+		recordTracksEvent
+	}
+)( localize( PluginInstallButton ) );
```
Comment (tyxla) [inline client/my-sites/plugins/plugin-install-button/index.jsx]: One new line at the end of the file should be enough. Currently I can see 2 of them.

File: client/my-sites/plugins/plugin-install-button/index.jsx [hunk 4]
```diff
@@ -1,116 +1,142 @@
...
+			return translate( 'Only the main site on a multi-site installation can install plugins.', {
+				args: { site: selectedSite.title }
 			} );
 		}
 
-		if ( ! this.props.selectedSite.canUpdateFiles && this.props.selectedSite.options.file_mod_disabled ) {
-			let reasons = utils.getSiteFileModDisableReason( this.props.selectedSite, 'modifyFiles' );
+		if ( ! selectedSite.canUpdateFiles && selectedSite.options.file_mod_disabled ) {
+			const reasons = utils.getSiteFileModDisableReason( selectedSite, 'modifyFiles' );
 			let html = [];
```
Comment (tyxla) [inline client/my-sites/plugins/plugin-install-button/index.jsx]: Arrays are essentially objects, and objects can be modified without reassigning or redefining the varaible; so this should be `const html = [];` instead.

**Comment by enejb** [comment]
Thanks for your help reviewing this Marin! 
I did a bit more testing and found some other issues. Which I fixed. 
Let me know if this still good to merge.

**Comment by tyxla** [comment]
LGTM - 🐑  it.

**Pull request merged**